src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunkReadSide.cs

```diff
@@ -208,7 +221,7 @@ private Midpoint[] PopulateMidpoints(int depth, ReaderWorkItem workItem) {
 					midpoints = new Midpoint[1 + (mapCount + segmentSize - 1) / segmentSize];
 				}
 
-				for (int x = 0, i = 0, xN = mapCount - 1; x < xN; x += segmentSize, i += 1) {
+				for (int x = 0, i = 0, xN = mapCount - 1; x < xN; x += segmentSize, i++, token.ThrowIfCancellationRequested()) {
```

Comment:
not sure if we ought to check the cancellation token this often,, maybe once at the top of the method would do